src/Service.Tests/SqlTests/GraphQLMutationTests/MultipleMutationIntegrationTests.cs

```diff
@@ -0,0 +1,429 @@
+            string createOneBookMutationName = "createbook";
+            string multipleCreateOneBookWithoutPublisher = @"mutation {
+                    createbook(item: { title: ""My New Book"" }) {
+                        id
+                        title
+                    }
+                }";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(
+                query: multipleCreateOneBookWithoutPublisher,
+                queryName: createOneBookMutationName,
+                isAuthenticated: true,
+                variables: null,
+                clientRoleHeader: "authenticated");
+
+            // Validate that an appropriate exception is thrown because the value for the referencing field
+            // publisher_id cannot be derived from any of the two possible sources mentioned above.
+            SqlTestHelper.TestForErrorInGraphQLResponse(
+                    actual.ToString(),
+                    message: "Missing value for required column: publisher_id for entity: Book at level: 1.",
+                    statusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
+                );
+        }
+
+        /// <summary>
+        /// Test to validate that for 'create multiple' mutations, we throw an appropriate exception when the user request provides
+        /// no source of value for a non-nullable and non-default referencing column.
+        /// There are 3 sources of values for a referencing column:
+        /// 1. Explicit value assignment to the referencing column,
+        /// 2. Value derived from the parent referenced entity,
+        /// 3. Providing value for inserting record in the referenced target entity, in which case value for referencing column
+        /// is derived from the value of corresponding referenced column after insertion in the referenced entity.
+        /// When the user provides none of the above, we throw an exception.
+        /// </summary>
+        [TestMethod]
+        public async Task AbsenceOfSourceOfTruthForReferencingColumnForCreateMultipleMutations()
+        {
+            // For a relationship between Book (Referencing) - Publisher (Referenced) defined as books.publisher_id -> publisher.id,
+            // in the request input:
+            // 1. No explicit value is provided for referencing field publisher_id
+            // 2. Book is the top-level entity - so there is no parent entity.
+            // 3. No value is assigned for the referenced entity via relationship field ('publisher').
+            string createmultipleBooksMutationName = "createbooks";
+            string createMultipleBooksWithoutPublisher = @"mutation {
+                    createbooks(items: [{ title: ""My New Book"" }]) {
+                        items{
+                           id
+                           title
+                        }
+                    }
+                }";
+
```

Comment:
need some tests that cover `createbooks(items:[{...},{...}])` where `items` has more than one item. This will theoretically further exercise the recursive validation functions.

Reply:
modified existing tests as that covers both.